File: reference/python/cli-evolution.py
from __future__ import annotations

import json
import sys
from pathlib import Path

from evolution import run_vectors


def main() -> None:
    if len(sys.argv) != 2:
        raise SystemExit("Usage: python reference/python/cli-evolution.py <manifest.json>")
    path = Path(sys.argv[1])
    manifest = json.loads(path.read_text(encoding="utf-8"))
    vectors = [json.loads((path.parent / (entry if isinstance(entry, str) else entry["file"])).read_text(encoding="utf-8")) for entry in manifest["vectors"]]
    print(json.dumps(run_vectors(vectors, manifest.get("profile", manifest.get("protocol", manifest.get("specVersion")))), sort_keys=True, separators=(",", ":")))


if __name__ == "__main__":
    main()

File: reference/typescript/src/cli-evolution.ts
import { readFileSync } from "node:fs";
import { dirname, join, resolve } from "node:path";
import { runEvolutionVectors, type EvolutionProfile } from "./evolution.js";

const input = process.argv[2];
if (input === undefined) {
  console.error("Usage: node dist/cli-evolution.js <manifest.json>");
  process.exitCode = 2;
} else {
  try {
    const path = resolve(input);
    const manifest = JSON.parse(readFileSync(path, "utf8")) as { protocol: EvolutionProfile; profile?: EvolutionProfile; vectors: readonly (string | { readonly file: string })[] };
    const document = { vectors: manifest.vectors.map((entry) => JSON.parse(readFileSync(join(dirname(path), typeof entry === "string" ? entry : entry.file), "utf8"))) };
    console.log(JSON.stringify(runEvolutionVectors(document, manifest.profile ?? manifest.protocol)));
  } catch (error) {
    console.error(error instanceof Error ? error.message : String(error));
    process.exitCode = 1;
  }
}
